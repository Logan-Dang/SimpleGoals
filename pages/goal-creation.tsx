import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Dimensions,
  TextInput,
} from 'react-native';

export default function NewGoal() {
  return <View style={styles.container}></View>;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    height: '100%',
    margin: 0,
  },

  title: {
    fontSize: 24,
  },
});
