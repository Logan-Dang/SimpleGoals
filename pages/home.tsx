import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Dimensions,
} from 'react-native';
export default function Home() {
  let currentTitle;
  let defaultSub;
  let backgroundImage = require('../assets/sunrise.png');
  let screenWidth = Dimensions.get('window').width;
  if (getTime() >= 6 && getTime() < 12) {
    currentTitle = 'Good Morning!';
    defaultSub = "Let's get started";
  } else if (getTime() >= 12 && getTime() <= 18) {
    currentTitle = 'Good Afternoon!';
    defaultSub = 'How was lunch?';
  } else {
    currentTitle = 'Good Evening!';
    defaultSub = "Let's review your day";
  }
  return (
    <View style={styles.container}>
      <ImageBackground
        source={backgroundImage}
        resizeMode='cover'
        style={styles.background}>
        <Text style={styles.title}>{currentTitle}</Text>
        <Text style={styles.subtitle}>{defaultSub}</Text>
        <Text>Goal 1</Text>
      </ImageBackground>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    height: '100%',
    margin: 0,
  },

  title: {
    fontSize: 24,
  },

  subtitle: {
    fontSize: 16,
    marginTop: '3%',
  },

  background: {
    alignItems: 'center',
    width: Dimensions.get('screen').width,
    height: Dimensions.get('screen').height,
    paddingTop: '15%',
    margin: 0,
  },
});

function getTime() {
  const currentTime = new Date();
  const time = currentTime.getHours();
  return time;
}
