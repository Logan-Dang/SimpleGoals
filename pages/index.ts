export { default as Home } from './home';
export { default as NewGoal } from './new-goal';
