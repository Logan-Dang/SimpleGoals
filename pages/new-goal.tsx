import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Dimensions,
  TextInput,
  KeyboardAvoidingView,
  findNodeHandle,
} from 'react-native';

import { Button, SomeText } from '../components';

import { FieldError } from 'react-hook-form';

export default function NewGoal() {
  return (
    <View>
      <TextInput style={styles.textStyle} value={'please work'} />
    </View>
  );
}

const styles = StyleSheet.create({
  textStyle: {
    marginTop: 200,
  },
});
