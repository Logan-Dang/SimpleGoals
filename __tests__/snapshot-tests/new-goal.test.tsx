import { create } from 'react-test-renderer';
import { NewGoal } from '../../pages';

it('renders correctly', () => {
  const tree = create(<NewGoal />).toJSON();
  expect(tree).toMatchSnapshot();
});
