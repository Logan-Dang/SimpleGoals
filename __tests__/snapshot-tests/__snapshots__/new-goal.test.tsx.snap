// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <RCTSafeAreaView
    accessible={true}
    emulateUnlessSupported={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Goal Name..."
      style={
        Object {
          "borderRadius": 6,
          "marginTop": 200,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Goal Description..."
      style={
        Object {
          "borderRadius": 6,
          "marginTop": 200,
        }
      }
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "JustifyContent": "center",
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "elevation": 3,
          "paddingHorizontal": 32,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Submit
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
